pytest/lib/branches.py

```diff
@@ -16,10 +16,19 @@
 _OUT_DIR = _REPO_DIR / 'target/debug'
 
 
-def current_branch():
-    return os.environ.get('BUILDKITE_BRANCH') or subprocess.check_output([
-        "git", "rev-parse", "--symbolic-full-name", "--abbrev-ref", "HEAD"
-    ]).strip().decode()
+def current_branch() -> str:
+    """Returns checked out branch name or sha if we’re on detached head."""
+    branch = os.environ.get('BUILDKITE_BRANCH')
```

Comment:
can you explain (in comment) - why we treat BUILDKITE in a special way?

Reply:
I mean, I wasn’t the one writing the original code and don’t really have any knowledge other than anyone who would read this code would have. There is a risk that whatever I comment will not be correct.